packages/react-ui-kit/src/Form/Select.tsx

```diff
@@ -19,240 +19,50 @@
 
 /** @jsx jsx */
 import {CSSObject, jsx} from '@emotion/react';
-
+import {useTheme} from '@emotion/react';
+import React, {FC, ReactElement} from 'react';
+import ReactSelect from 'react-select';
+import {StylesConfig} from 'react-select/dist/declarations/src/styles';
+import {StateManagerProps} from 'react-select/dist/declarations/src/useStateManager';
+
+import {customStyles} from './SelectStyles';
+import {CustomOption, DropdownIndicator, IndicatorsContainer, Menu, ValueContainer} from './SelectComponents';
 import type {Theme} from '../Layout';
-import {filterProps} from '../util';
-import {inputStyle} from './Input';
-import React, {ReactElement, useEffect, useRef, useState} from 'react';
 import InputLabel from './InputLabel';
 
-export type SelectOption = {
+export type Option = {
   value: string | number;
   label: string;
   description?: string;
+  isDisabled?: boolean;
 };
 
-export interface SelectProps<T extends SelectOption = SelectOption> {
+interface CustomSelectProps extends StateManagerProps {
   id: string;
-  onChange: (selectedOption: T['value']) => void;
   dataUieName: string;
-  options: T[];
```

Comment:
i know StateManager adds a lot of these props but my concern is that we lose some readability of our API typing if we don't also define options, onChange, value etc. 
Especially since our options now has a specific type expected options: Option[]

WDYT?

Reply:
Are these optional values in the state manager? If yes they shouldn't be in there and instead be typed at the component where they are needed.
Not a big fan of state managers that include everything eventually. I prefer strongly typed components where I can see what they need directly.

Reply:
that would be my thought as well, it makes our components unreadable to anyone who isn't fully versed on the state manager. (in this case the state manager is coming from react-select). 
As this in a component library we should tell people what params our components can use.